syncopy/tests/test_continuousdata.py

```diff
@@ -935,3 +923,216 @@ def test_sd_parallel(self, testcluster):
+            dummy3 = load(fname)
+            for attr in checkAttr:
+                assert np.array_equal(getattr(dummy3, attr), getattr(dummy, attr))
+            save(dummy3, container=os.path.join(tdir, "ymmud"))
+            dummy4 = load(os.path.join(tdir, "ymmud"))
+            for attr in checkAttr:
+                assert np.array_equal(getattr(dummy4, attr), getattr(dummy, attr))
+            del dummy, dummy3, dummy4  # avoid PermissionError in Windows
+
+            # ensure trialdefinition is saved and loaded correctly
+            dummy = CrossSpectralData(self.data, samplerate=1000)
+            dummy.trialdefinition = self.trl
+            dummy.save(fname, overwrite=True)
+            dummy2 = load(filename)
+            assert np.array_equal(dummy.trialdefinition, dummy2.trialdefinition)
+
+            # test getters
+            assert np.array_equal(dummy.sampleinfo, dummy2.sampleinfo)
+            assert np.array_equal(dummy._t0, dummy2._t0)
+            assert np.array_equal(dummy.trialinfo, dummy2.trialinfo)
+
+            # swap dimensions and ensure `dimord` is preserved
+            dummy = CrossSpectralData(self.data, dimord=["freq", "channel_j", "channel_i", "time"],
+                                      samplerate=1000)
+            dummy.save(fname + "_dimswap")
+            filename = construct_spy_filename(fname + "_dimswap", dummy)
+            dummy2 = load(filename)
+            assert dummy2.dimord == dummy.dimord
+            assert dummy2.channel_i.size == self.nci # swapped
+            assert dummy2.channel_j.size == self.nf # swapped
+            assert dummy2.freq.size == self.ns  # swapped
+            assert dummy2.data.shape == dummy.data.shape
+
+            # Delete all open references to file objects b4 closing tmp dir
+            del dummy, dummy2
+
+    # test data-selection via class method
+    def test_csd_dataselection(self):
+
+        # Create testing objects (regular and swapped dimords)
+        dummy = CrossSpectralData(data=self.data,
+                                  samplerate=self.samplerate)
+        dummy.trialdefinition = self.trl
+        ymmud = CrossSpectralData(data=np.transpose(self.data, [3, 2, 1, 0]),
+                                  samplerate=self.samplerate,
+                                  dimord=CrossSpectralData._defaultDimord[::-1])
+        ymmud.trialdefinition = self.trl
+
+        for obj in [dummy, ymmud]:
+            idx = [slice(None)] * len(obj.dimord)
+            timeIdx = obj.dimord.index("time")
+            chanIdx = obj.dimord.index("channel_i")
+            chanJdx = obj.dimord.index("channel_j")
+            freqIdx = obj.dimord.index("freq")
+            for trialSel in trialSelections:
+                for chaniSel in chanSelections:
+                    for chanjSel in chanSelections:
+                        for timeSel in timeSelections:
+                            for freqSel in freqSelections:
```

Comment:
Here `itertools.product` might reduce some for-loops..?!

Reply:
Hm, interesting idea. It would definitely reduce the number of for loops but I'm not sure it would add anything in terms of legibility or performance (the lion share of computational time is taken up by the actual superlet computation, I don't think inefficiencies due to nested for loops are a concern here). Since the code is functional and we're on a relatively tight schedule I vote for leaving it as it is.